lib/spack/spack/cmd/__init__.py

```diff
@@ -308,7 +308,10 @@ def get_arg(name, default=None):
 
             for abbrv, spec in zip(abbreviated, specs):
                 prefix = gray_hash(spec, hlen) if hashes else ''
-                print(prefix + (format % (abbrv, spec.prefix)))
+                if spec.external_path:
+                    print(format % (abbrv, spec.external_path))
```

Comment:
Thanks for submitting this! One question: is there a reason the prefix is omitted for `external_path`?

Reply:
In my experience, `prefix` has always been empty for external packages and `external_path` has contained the absolute path to every package. I assume this is an implementation decision that has already exited for good reasons, but I could be mistaken and if prefix should contain a path in specific scenarios I will gladly re-submit a PR. Thank you.